components/formats-gpl/src/loci/formats/in/KLBReader.java

```diff
@@ -0,0 +1,577 @@
+    for (int i=0; i < listOfFiles.length; i++) {
+      String fileName = listOfFiles[i].getName();
+      if (fileName.startsWith(basePrefix)) {
+        int currentTimepoint = DataTools.parseInteger(fileName.substring(fileName.indexOf(".TM")+3, fileName.indexOf("_timeFused")));
+        File[] innerFileList = listOfFiles[i].listFiles();
+        for (int j=0; j < innerFileList.length; j++) {
+          String innerFileName = innerFileList[j].getName();
+          if (innerFileName.contains("fusedStack")) {
+            int currentChannelNum = DataTools.parseInteger(innerFileName.substring(innerFileName.indexOf("_CHN")+4, innerFileName.indexOf('.')));
+            if (innerFileName.indexOf("fusedStack_") >= 0) {
+              String projection = innerFileName.substring(innerFileName.indexOf("fusedStack_")+11, innerFileName.indexOf("Projection"));
+              if (projection.equals("xy")) {
+                xyProjections[currentTimepoint][currentChannelNum] = innerFileList[j].getAbsolutePath();
+                if (currentTimepoint == 0 && currentChannelNum == 0) {
+                  in.close();
+                  in = new RandomAccessInputStream(innerFileList[j].getAbsolutePath());
+                  readHeader(core.get(0));
+                }
+              }
+              else if (projection.equals("xz")) {
+                xzProjections[currentTimepoint][currentChannelNum] = innerFileList[j].getAbsolutePath();
+                if (currentTimepoint == 0 && currentChannelNum == 0) {
+                  in.close();
+                  in = new RandomAccessInputStream(innerFileList[j].getAbsolutePath());
+                  readHeader(core.get(1));
+                }
+              }
+              else if (projection.equals("yz")) {
+                yzProjections[currentTimepoint][currentChannelNum] = innerFileList[j].getAbsolutePath();
+                if (currentTimepoint == 0 && currentChannelNum == 0) {
+                  in.close();
+                  in = new RandomAccessInputStream(innerFileList[j].getAbsolutePath());
+                  readHeader(core.get(2));
+                }
+              }
+            }
+            else {
+              fusedStacks[currentTimepoint][currentChannelNum] = innerFileList[j].getAbsolutePath();
+              if (currentTimepoint == 0 && currentChannelNum == 0) {
+                in.close();
+                in = new RandomAccessInputStream(innerFileList[j].getAbsolutePath());
+                readHeader(core.get(3));
+              }
+            }
+          }
+        }
+      }
+    }
+    String imageID = MetadataTools.createLSID("Image", 0);
+    store.setImageID(imageID, 0);
+    imageID = MetadataTools.createLSID("Image", 1);
+    store.setImageID(imageID, 1);
+    imageID = MetadataTools.createLSID("Image", 2);
+    store.setImageID(imageID, 2);
+    imageID = MetadataTools.createLSID("Image", 3);
+    store.setImageID(imageID, 3);
+    MetadataTools.populatePixels(store, this);
```

Comment:
```populatePixels``` automatically calls ```setImageID``` for each series, so I would expect that lines 386-393 could be removed.